### ISSUE TITLE
Process oriented style DAG building in nested workflow

### ISSUE THREAD

**Issue by gaow** [vatlab/SoS#1086]
Separated from a discussion elsewhere, related to DAG: if i do have defined input and output for, 

```
[method_1]
input: ...
output: ...

[method_2]
input: ...
output: ...

[another_method_1]
input: ...
output: ...

[another_method_2]
input: ...
output: ...

[default]
sos_run('method')
sos_run('another_method')
```

then in `default` the two `sos_run` will work in parallel, right? This is in fact exactly our process oriented example. I seem to recall at some point we have a `no_wait` option to `sos_run` to do this manually -- that's in case of `input` and `output` are not defined? In this case parallelization should be automatic?

and from @BoPeng 

> I played around with the idea for a while and found it doable, but still more complicated than I thought. Basically, anything touching the core mesh of messaging can cause unexpected outcome. I would keep this ticket open and see if I can get a chance to revisit.

**Comment by BoPeng** [comment]
No. The two `sos_run` are executed sequentially as a Python block. It is possible to add an option to it, something like

```
[default]
sos_run('method', concurrent=True)
sos_run('another_method', concurrent=True)
```
but I could not make it to work correctly given the possibilities that nested workflows can have nested workflows, tasks etc. As I said, it could work but it is not a priority for me now.

**Comment by gaow** [comment]
Resolved in #1167 with restrictions.
